tests/form.js

```diff
@@ -527,7 +527,61 @@ define([
                     }
                 });
             });
+        });
+
+        describe("hashtag logic", function () {
+            var form;
+            before(function () {
+                form = util.loadXML("");
+            });
+
+            _.each({
+                "": null,
+                "#": null,
+
+                "#form": null,
+                /* TODO should these pass? (they don't)
+                "#form/": "is",
+                "#form/prop": "has",
+                */
+
+                "/data": null,
+                "/data/": null,
+                "/data/prop": "xpath",
+
+                "#case": null,
+                "#case/": "is",
+                "#case/prop": "has xpath",
+                /* TODO should these pass? (they don't)
+                "#case/ ": null,
+                "#case/+": null,
```

Comment:
Agree `#case/+` should be `null`. I think `#case/: "is"` makes sense, why do you think it should be `null`?